draft-ietf-privacypass-architecture.md

```diff
@@ -420,12 +424,55 @@ Therefore, we are careful to ensure that the number of servers is kept
 strictly bounded. The actual servers can be replaced with different
 servers as long as the total never exceeds this bound. Moreover, server
 replacements also have an effect on client anonymity that is similar to
-when a key rotation occurs. server so replacement should only be
-permitted at similar intervals.
+when a key rotation occurs.
 
 See {{privacy}} for more details about maintaining privacy with multiple
 servers.
 
+# Metadata
+
+Certain instantiations of Privacy Pass may permit public or private
+metadata to be cryptographically bound to a token. As an example, one
+trivial way to include public metadata is to assign a unique issuer
+public key for each value of metadata, such that N keys yields log2(N)
+bits of metadata. The total amount of metadata bits included in a token
+is the sum of public and private metadata bits. See {{parametrization}}
+for discussion about metadata limits.
+
+Public metadata is that which clients can observe as part of the token
+issuance flow. Public metadata can either be transparent or opaque. For
+example, transparent public metadata is a value that the client either
+generates itself, or the server provides during the issuance flow and
+the client can check for correctness. Opaque public metadata is metadata
+the client can see but cannot check for correctness. As an example, the
+opaque public metadata might be a "fraud detection signal", computed on
+behalf of the server, during token issuance. In normal circumstances,
+clients cannot determine if this value is correct or otherwise a tracking
+vector.
+
+Private metadata is that which clients cannot observe as part of the token
+issuance flow. In {{draft-davidson-pp-protocol}}, it is possible to include
+private metadata to redemption tokens. The core protocol instantiation that
+is described does not include additional metadata. However, future instantiations
+may use this functionality to provide redemption verifiers with additional
+information about the user.
+
```

Comment:
It might be worth separating the last two paragraphs as they relate mostly to privacy considerations. What do you think?

```suggestion

## Client privacy implications

```

Reply:
Yeah, good call :-)